packages/dynamodb/src/index.ts

```diff
@@ -1,447 +1,436 @@
-import { createHash, randomBytes } from "crypto"
-import { Profile, Session, User } from "next-auth"
-import { Adapter } from "next-auth/adapters"
+import { BatchWriteCommandInput, DynamoDBDocument } from "@aws-sdk/lib-dynamodb"
+import { randomBytes } from "crypto"
+import { Account, User } from "next-auth"
+import type {
+  Adapter,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "next-auth/adapters"
+
+// internal interfaces for data as is it saved in dynamodb
+// adds some dynamodb specific types such as index keys (pk, sk, GSI1PK, GSI1SK)
+// and changes date types to number for expires property (epoch time saved in seconds) or string (IsoString for emailVerfied prop)
+interface AdapterSessionDynamo {
+  pk: string
+  sk: string
+  id: string
+  /** A randomly generated value that is used to get hold of the session. */
+  sessionToken: string
+  /** Used to connect the session to a particular user */
+  userId: string
+  /** This property can be used to set up dynamodb TTL function see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html for more information */
+  expires: number
+  GSI1PK?: string
+  GSI1SK?: string
+  type: "SESSION"
+}
+interface VerificationTokenDynamo {
+  pk: string
+  sk: string
+  identifier: string
+  /** This property can be used to set up dynamodb TTL function see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html for more information */
+  expires: number
+  token: string
+  type: "VR"
+}
+
+interface AdapterUserDynamo extends User {
+  pk: string
+  sk: string
+  id: string
+  emailVerified: string | null
+  GSI1PK?: string
+  GSI1SK?: string
+  type: "USER"
+  email: string
+}
```

Comment:
These all should be the extension of the corresponding `next-auth` interfaces.

Remove any of the fields that already exist.

Reply:
removed those types and replaced with a generic type